cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -383,16 +388,54 @@ func ValidateIPNetFromString(subnet string, minAddrs int64, fldPath *field.Path)
 	return allErrs
 }
 
-// ValidateNetworking validates networking configuration
+// ValidateIPNetFromStringForDualStack validates network portion of ip addresses for dual-stack configurations
+func ValidateIPNetFromStringForDualStack(subnet string, minAddrs int64, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	subnets := strings.Split(subnet, ",")
+	if len(subnets) < 2 {
```

Comment:
[2]
i think that instead of having a separate function `ValidateIPNetFromStringForDualStack` we can just handle dual stack checks in the following fashion:

- in `ValidateNetworking` we detect if feature gate is enabled
- we pass that as `bool` to `ValidateIPNetFromString`
- `ValidateIPNetFromString` checks if the subnet string is a list and if the feature gate is not enabled error out?
(it seems that we aren't checking the feature gate in validation.go).

this makes `ValidateIPNetFromString` handle both lists and single CIDRs.
and new tests can be added in `TestValidateIPNetFromString`.

WDYT?

Reply:
To detect if FG is enabled in `ValiateNetworking` we will need `ClusterConfiguration` passed as arg. That's not ideal IMO. 
I initially went with a very similar approach of passing in a `bool` to `ValidateNetworking` and down to `ValidateIPNetFromString`; I didn't like it b/c it bloated `ValiateIPNetFromString` and put dualstack logic in the top `ValidateClusterConfiguration`. The current approach seems a little cleaner. 
I'm fine to change it though. What do others think?

Reply:
I have some concern about this approach...
What if someone calls `ValidateNetworking` directly? it will always assume `isDualStackNetworkingConfig == false`

Reply:
+1 @neolit123 - I'd prefer us checking  the feature gates on `ValidateNetworking` and passing it down. also, using a global variable is not ideal.